subnet/deployment-generator/src/config_gen.js

```diff
@@ -6,25 +6,25 @@ const { off } = require('process');
 dotenv.config({ path: `${__dirname}/gen.env` });
 // console.log(__dirname)
 
-var config = {
+let config = {
   deployment_path:    (process.env.CONFIG_PATH            || ''),
   num_machines:       parseInt(process.env.NUM_MACHINE),
   num_subnet:         parseInt(process.env.NUM_SUBNET),
   ip_1:               (process.env.MAIN_IP                || ''),
   network_name:       (process.env.NETWORK_NAME),
   network_id:         parseInt(process.env.NETWORK_ID     || Math.floor(Math.random() * (65536 - 1) + 1)),
   secret_string:      (process.env.SERVICES_SECRET        || crypto.randomBytes(10).toString('hex')),
-  relayer_mode:       (process.env.RELAYER_MODE           || 'full'), //full or lite
+  relayer_mode:       (process.env.RELAYER_MODE           || 'full'), //full or lite //in upgradable csc both are deployed
   docker_image_name:  (process.env.IMAGE_NAME             || 'xinfinorg/subnet-generator:latest'),
   operating_system:   (process.env.OS                     || 'linux'),
   version: {
     subnet:   (process.env.VERSION_SUBNET   || 'v0.2.1'),
     bootnode: (process.env.VERSION_BOOTNODE || 'v0.2.1'),
-    observer: (process.env.VERSION_OBSERVER || 'latest'),
-    relayer:  (process.env.VERSION_RELAYER  || 'v0.2.1'),
+    // observer: (process.env.VERSION_OBSERVER || 'latest'),
+    relayer:  (process.env.VERSION_RELAYER  || 'v0.2.2'),
     stats:    (process.env.VERSION_STATS    || 'v0.1.8'),
     frontend: (process.env.VERSION_FRONTEND || 'v0.1.8'),
-    // csc:      (process.env.VERSION_CSC      || 'v0.1.1'),
+    csc:      (process.env.VERSION_CSC      || 'v0.1.1'),
     // zero:     (process.env.VERSION_ZERO     || 'v0.1.1')
```

Comment:
put latest tag is better, you won't need to change it again and again.

Reply:
I think latest will cause many different version groupings and they are not guaranteed to be compatible.  In the current way we can control official release versions and only refer to the latest generator. Also latest causes issues with checking current version or accidental updates with docker pull.

Reply:
hmm dependency versioning... I got you now.